packages/superset-ui-encodable/src/fillers/expandLabelOverlapStrategy.ts

```diff
@@ -0,0 +1,29 @@
+import { LabelOverlapStrategy, LabelOverlapType } from '../types/Axis';
+import { ChannelType } from '../types/Channel';
+import { isX } from '../typeGuards/Channel';
+
+const STRATEGY_FLAT = { strategy: 'flat' } as const;
+const STRATEGY_ROTATE = { labelAngle: 40, strategy: 'rotate' } as const;
+
+export default function expandLabelOverlapStrategy(
+  channelType: ChannelType,
+  labelOverlap: LabelOverlapType = 'auto',
+): LabelOverlapStrategy {
+  let output: LabelOverlapStrategy;
```

Comment:
any reason not to just return in the `switch`?

Reply:
I was debating whether to add cloning (`{...output}`) to every return or add once at the end.